src/components/Table/table-cells/TableBodyCell.tsx

```diff
@@ -15,16 +14,67 @@ export const TableBodyCell: React.FC<TableCellProps> = ({
     className,
     iconStart,
     iconEnd,
-    isFirstColumn,
-    canExpand,
-    isExpanded,
-    depth,
-    getToggleRowExpandedProps,
+    expandableColumnIndex,
+    row,
+    cell,
     children,
     ...props
 }) => {
-    const paddingLeft = `${depth * 0.75}rem`;
-    const marginLeft = spacings.spacingMedium;
+    // Fix this type definition of any
+    const isExpandableColumn = (row as any)?.cells[expandableColumnIndex].value === cell?.value;
+    const leftMarginBuffer = 20; // px
+    const leftMarginIndentDepth = row?.depth * 20; // px
+
+    const renderIconTextIcon = () => (
+        <IconTextIcon
+            component="div"
+            iconStart={iconStart}
+            typographyProps={{isNowrap: true}}
+        >
+            {children}
+        </IconTextIcon>
+    );
+
+    const renderTableCell = () => {
+        // These are cells that are in the expandable row (canExpand) and it is the column in
+        // which the cells show the chevron icon to expand and collapse sub-rows (isExpandableColumn)
+        if (isExpandableColumn && row?.canExpand) {
+            return (
+                <TableCell
+                    {...
+                        (row?.canExpand && isExpandableColumn
+                            ? row?.getToggleRowExpandedProps({style: {marginLeft: `${leftMarginIndentDepth}px`}})
+                            : {}
+                        )
+                    }
+                >
+                    {row?.isExpanded
+                        ? <ChevronDown className="moonstone-marginRightNano"/>
+                        : <ChevronRight className="moonstone-marginRightNano"/>
+                    }
+                    {renderIconTextIcon()}
+                </TableCell>
+            );
+        }
+
+        // These are cells which are in the expandable column (isExpandableColumn), but themselves
+        // do now have sub-rows. Therefore, they need to have the appropriate nested indentation,
+        // but do not have the chevron to expand/collapse rows underneath them.
+        // Also, a buffer of 20px is added so that they are aligned with the cells that do have
+        // the chevron icons for expand/collapse
+        if (isExpandableColumn && !row?.canExpand) {
+            return (
+                <TableCell style={{marginLeft: `${leftMarginIndentDepth + leftMarginBuffer}px`}}>
+                    {renderIconTextIcon()}
+                </TableCell>
+            );
+        }
```

Comment:
The cells without children at the root level are also affected and they are not aligned with the header anymore. I think we also need to check the depth to add the `leftMarginBuffer`.

```suggestion
        if (isExpandableColumn && !row?.canExpand && row?.depth > 1) {
            return (
                <TableCell style={{marginLeft: `${leftMarginIndentDepth + leftMarginBuffer}px`}}>
                    {renderIconTextIcon()}
                </TableCell>
            );
        }
```

Reply:
I actually wanted cells without children at the root level affected by this as well because we need room to show the expansion chevron on the left of the content inside the cell. I think it would be confusing if the the content part of all the root level cells didn't line up. And, I don't think we should push them to the left because we can't rely on there being extra space on the left side of the Table (in the case that there is other column to the left of the expandable column).

Reply:
I understand